packages/presentation/lib/src/base_bloc/bloc.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:presentation/src/navigation/app_navigator.dart';
+import 'package:presentation/src/navigation/base_arguments.dart';
+import 'base_tile.dart';
+
+abstract class Bloc<T extends BaseArguments, D> {
+  Stream<BaseTile<D?>> get dataStream;
+
+  D get tile;
+
+  void init();
+
+  void initArgs(T args);
+}
+
+abstract class BlocImpl<T extends BaseArguments, D> implements Bloc<T, D> {
+  final _dataStream = StreamController<BaseTile<D?>>();
+  BaseTile<D> baseTile = BaseTile.init();
+  D _blocTile;
+
+  @protected
+  final appNavigator = GetIt.I.get<AppNavigator>();
+
+  BlocImpl(this._blocTile);
+
+  @override
+  D get tile => _blocTile;
```

Comment:
why do you use _blocTile for?

Reply:
to have access to tile in my bloc class, like in library bloc

Reply:
for what? You have access to take in your block screen(Main Block,Splash Block) and your screen also have access to tile because streambuilder return tile

Reply:
I prefer to use in bloc this construction which needed tile in bloc:
 emit(ile.copyWith())
 additionally, in guessNumber was case with verification of the textField. StreamBuilder return tile with some delay, and verification executes with previous tile. So I used:
validator: (_) => bloc.tile.validateResult
instead of:
validator: (_) => tile.validateResult,